Sync/BookmarksDownloader.swift

```diff
@@ -230,10 +244,24 @@ class BatchingDownloader<T: CleartextPayloadJSON> {
             return deferMaybe(.NoNewData)
         }
 
-        return self.downloadNextBatchWithLimit(limit, advancingOnCompletionTo: modified)
+        return self.downloadNextBatchWithLimit(limit, infoModified: modified)
+    }
+
+    // We're either fetching from our current base timestamp with no offset,
+    // or the timestamp we were using when we last saved an offset.
+    func fetchParameters() -> (String?, Timestamp) {
+        if let (offset, since) = self.nextFetchParameters {
```

Comment:
actually a `guard` would be better here if we're going to be all swift 2.0-y

` guard let (offset, since) = self.nextFetchParameters else { return (nil, max(self.lastModified, self.baseTimestamp)) }
return (offset, since)`